master/internal/db/postgres_checkpoints.go

```diff
@@ -0,0 +1,86 @@
+package db
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+// FilterForRegisteredCheckpoints gets the deleted checkpoints provided in the model registry.
+func (db *PgDB) FilterForRegisteredCheckpoints(deleteCheckpoints []uuid.UUID) ([]uuid.UUID, error) {
+	var checkpointIDRows []struct {
+		ID uuid.UUID
+	}
+
+	if err := db.queryRows(`
+	SELECT DISTINCT(c.uuid) AS ID FROM checkpoints_view AS c 
```

Comment:
we don't nede to join at all here, we can just `SELECT DISTINCT(mv.checkpoint_uuid) FROM model_versions AS mv WHERE mv.checkpoint_uuid IN (SELECT UNNEST($1::uuid[]))`.

Reply:
oops yes! not sure why i used join, thank you!